advanced/metrics.yaml

```diff
@@ -0,0 +1,187 @@
+      type: object
+      properties:
+        flow_names:
+          type: array
+          items:
+            type: string
+            x-constraint:
+            - '/components/schemas/Flow/properties/name'
+            example: ['Tx Port -> Rx Port']
+        event_sources:
+          type: array
+          items:
+            type: string
+            x-constraint:
+            - '/components/schemas/Device.BgpRoute/properties/name'
+            example: ['BGP Route Range 1']
+
+    AdvancedMetrics.Response:
+      type: object
+      properties:
+        event_metrics:
+          type: array
+          items:
+            $ref: './event.yaml#/components/schemas/AdvancedMetrics.Event'
+        latency_metrics:
+          type: array
+          items:
+            $ref: '#/components/schemas/AdvancedMetrics.Latency'
+        convergence_metrics:
+          type: array
+          items:
+            $ref: '#/components/schemas/AdvancedMetrics.Convergence'
+
+    AdvancedMetrics.Latency:
+      description: |-
+        The container for latency metrics. The min/max/avg values are dependent 
+        on the type of latency measurement mode that is configured.
+      type: object
+      properties:
+        flow_name:
+          description: |-
+            The name of the source.
+            Flow + metric group?
+          type: string
+          example: 'Tx Port -> Rx Port'
+        minimum_ns:
+          description: |-
+            Minimum latency in nanoseconds
+          type: number
+          format: double
+        maximum_ns:
+          description: |-
+            Maximum latency in nanoseconds
+          type: number
+          format: double
+        average_ns:
+          description: |-
+            Average latency in nanoseconds
+          type: number
+          format: double
+
+    AdvancedMetrics.Convergence:
+      description: |-
+        The container for data plane and control plane convergence metrics
+      type: object
+      properties:
+        flow_name:
+          description: |-
+            The name of the flow that is affected by the trigger event.
+          type: string
+          x-constraint:
+          - '/components/schemas/Flow/properties/name'
+          example: 'Tx Port -> Rx Port'
+        event_source_name:
+          description: |-
+            The name of the BGP IPv4/IPv6 route that produced the trigger event.
+          type: string
+          x-constraint:
+          - '/components/schemas/Device.BgpRoute/properties/name'
+          example: 'BGP Route Range 1'
+        event:
+          description: |-
+            The type of event that triggered the BGP convergence.
+          type: string
+          enum:
+          - route_advertise
+          - route_withdraw
+          example: 'route_withdraw'
+        event_begin_ns:
+          description: |-
+            The timestamp of when the event began.
+          type: number
+          format: double
+        event_end_ns:
```

Comment:
What does `event_end_ns` imply ? e.g. if somebody triggers `route_withdraw`, `event_begin_ns` will consist of the timestamp when it was triggered. But what would `event_end_ns` contain ?

Reply:
if the user is interested in data_plane_convergence, should they just ask for 
AdvancedMetrics.DataplaneConvergence.time

and AdvancedMetrics.DataplaneConvergence.details for the detailed events/timestamps that are used to arrive at the "time"?

CPDP will then be
AdancedMetrics.ControlplaneDataplaneConvergence ?  

this will probably make the known "events" being used to arrive at the time be more specific to the specific methodology of interest?